src/main/java/com/example/demo/service/impl/MessageServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.example.demo.service.impl;
+
+import com.example.demo.model.dto.MessageDto;
+import com.example.demo.model.dto.MessageDtoFactory;
+import com.example.demo.model.entity.Message;
+import com.example.demo.repository.MessageRepository;
+import com.example.demo.service.MessageService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class MessageServiceImpl implements MessageService {
+
+    private final MessageRepository messageRepository;
+
+    @Override
+    @Transactional
+    public void save(Message message) {
+        message.setCurrentTime(new Date().getTime());
+
+        messageRepository.save(message);
+    }
+
+    @Override
+    @Transactional
+    public List<Message> getAllMessage() {
+        List<Message> messages = messageRepository.findAll();
+        return messages;
+    }
+
+    @Override
+    @Transactional
+    public List<MessageDto> getAllMessageDto() {
+        List<Message> messages = messageRepository.findAll();
+        List<MessageDto> messageDtoList = new ArrayList<>();
```

Comment:
In such case Stream API is very powerful tool.

Reply:
Done